tests/wpt/tests/websockets/opening-handshake/received-301-code.html

```diff
@@ -0,0 +1,23 @@
+<!doctype html>
+<title>WebSockets: 301 Code Failed</title>
+<meta name=author title="Domenico Rizzo" href="mailto:domenico.rizzo@gmail.com">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="../constants.sub.js"></script>
+<div id="log"></div>
+<script>
+async_test(function(t) {
+    t.step(function() {
+        const ws = new WebSocket(SCHEME_DOMAIN_PORT + '/301');
```

Comment:
This is a good start! The 301_wsh.py websocket handler isn't exactly what we want; instead, instead we can use a [python HTTP handler](http://web-platform-tests.org/writing-tests/python-handlers/index.html#example-dynamic-http-headers) to return the 301 code as part of the initial response. I verified that this works locally:
```py
def main(request, response):
    headers = [('Location', '/echo')]
    return (301, "moved"), headers, ''
```
and then we can use `new WebSocket(new URL("redirect_response.py", location.href))`

Reply:
Did you forget to commit changes related to this comment?

Reply:
I haven't finished yet, I pushed wrongly.